src/prepare_data.py

```diff
@@ -0,0 +1,45 @@
+import numpy as np
+import pandas as pd
+from torch.utils.data import Dataset, DataLoader
+import os
+from utils import get_spectrograms
+import hyperparameters as hp
+import librosa
+
+class PrepareDataset(Dataset):
+    """LJSpeech dataset."""
+
+    def __init__(self, csv_file, root_dir):
+        """
+        Args:
+            csv_file (string): Path to the csv file with annotations.
+            root_dir (string): Directory with all the wavs.
+
+        """
+        self.landmarks_frame = pd.read_csv(csv_file, sep='|', header=None)
+        self.root_dir = root_dir
+
+    def load_wav(self, filename):
+        return librosa.load(filename, sr=hp.sample_rate)
+
+    def __len__(self):
+        return len(self.landmarks_frame)
+
+    def __getitem__(self, idx):
+        wav_name = os.path.join(self.root_dir, self.landmarks_frame.ix[idx, 0]) + '.wav'
+        mel, mag = get_spectrograms(wav_name)
+
+        np.save(wav_name[:-4] + '.pt', mel)
+        np.save(wav_name[:-4] + '.mag', mag)
+
+        sample = {'mel':mel, 'mag': mag}
+
+        return sample
+
+if __name__ == '__main__':
+    dataset = PrepareDataset(os.path.join(hp.data_path,'metadata.csv'), os.path.join(hp.data_path,'wavs'))
+    dataloader = DataLoader(dataset, batch_size=1, drop_last=False, num_workers=8)
+    from tqdm import tqdm
+    pbar = tqdm(dataloader)
+    for d in pbar:
```

Comment:
Is this just so __getitem__ gets called on each wav? If so, is it really necessary to wrap this in a Pytorch DataLoader/Dataset object?

Reply:
Yes, it is not. This is just copied over from the old repo. The code is terrible but there's not too much value in refactoring this code. It is only called once to make the mel files before training on a single machine.

Reply:
I think the only other reason they use DataLoader is so that there are multiple num_worker sub-processes to speed it up.

This looks good, I used this before the dataset as well.